libraries/plugins/tags/include/scorum/tags/tags_api_impl.hpp

```diff
@@ -470,6 +517,11 @@ class tags_api_impl
         if (posts.empty())
             return result;
 
+        if (!query.exclude_tags.empty())
+            posts = exclude(posts, query, tag_filter);
```

Comment:
You are already checking posts collection for emptyness inside `exclude` function so no need to check this twice. Moreover it is better to move this line `posts = exclude(posts, query, tag_filter);` before 
```
if (posts.empty())
    return result;
```